src/Altinn.Common/Altinn.EFormidlingClient/Altinn.EFormidlingClient/EFormidlingClient.cs

```diff
@@ -0,0 +1,381 @@
+                responseBody = await response.Content.ReadAsStringAsync();
+                Conversation conversation = JsonConvert.DeserializeObject<Conversation>(responseBody);
+                _logger.LogDebug(responseBody);
+
+                return conversation;
+            }
+            catch (HttpRequestException e)
+            {
+                _logger.LogError("Message :{Exception} ", e.Message);
+                throw;
+            }
+
+        }
+
+        /// <inheritdoc/>
+        public async Task<Conversation> GetConversationByMessageId(string id)
+        {
+            string responseBody;
+
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException();
+            }
+
+            try
+            {
+                HttpResponseMessage response = await _client.GetAsync($"conversations/messageId/{id}");
+                responseBody = await response.Content.ReadAsStringAsync();
+                Conversation conversation = JsonConvert.DeserializeObject<Conversation>(responseBody);
+                _logger.LogDebug(responseBody);
+
+                return conversation;
+            }
+            catch (HttpRequestException e)
+            {
+                _logger.LogError("Message :{Exception} ", e.Message);
+                throw;
+            }
+
+        }
+
+        /// <inheritdoc/>
+        public async Task<Statuses> GetMessageStatusById(string id)
+        {
+            string responseBody;
+
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException();
+            }
+
+            try
+            {
+                HttpResponseMessage response = await _client.GetAsync($"statuses?messageId={id}");            
+                responseBody = await response.Content.ReadAsStringAsync();
+                Statuses status = JsonConvert.DeserializeObject<Statuses>(responseBody);
+                _logger.LogDebug(responseBody);
+
+                return status;
+            }
+            catch (HttpRequestException e)
+            {
+                _logger.LogError("Message :{Exception} ", e.Message);
+                throw;
+            }       
+        }
+
+        /// <inheritdoc/>
+        public async Task<bool> UploadAttachment(Stream stream, string id, string filename)
+        {
+            if (stream == null || string.IsNullOrEmpty(id) || string.IsNullOrEmpty(filename))
+            {
+                throw new ArgumentNullException();
+            }
+
+            var streamContent = new StreamContent(stream);
+            streamContent.Headers.ContentDisposition = new ContentDispositionHeaderValue("attachment");
+            streamContent.Headers.ContentDisposition.Name = "attachment";
+            streamContent.Headers.ContentDisposition.FileName = filename;
+            streamContent.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+          
```

Comment:
Wondering if it's better if these are only set if not defined in the incomming stream? As we have accurate contet types available in the app we could use those and only sett to octet-stream if they aren't defined?